contracts/core/GlobalReentrancyLock.sol

```diff
@@ -0,0 +1,116 @@
+pragma solidity 0.8.13;
+
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import {PermissionsV2} from "./PermissionsV2.sol";
+import {IGlobalReentrancyLock} from "./IGlobalReentrancyLock.sol";
+
+/// @notice inpsired by the openzeppelin reentrancy guard smart contracts
+/// data container size has been changed
+/// @dev allows contracts with the GLOBAL_LOCKER_ROLE to call in and lock and unlock
+/// this smart contract.
+/// Governor can unpause if locked but not unlocked.
+contract GlobalReentrancyLock is IGlobalReentrancyLock, PermissionsV2 {
```

Comment:
I don't really like that this has to inherit `PermissionsV2` and is tightly coupled to the permissions contract. A reentrancy lock feels like it should be a standalone functional piece of software and Core should inherit seperately from the lock and permissions. 

Not much you can do though, as you need this modifier and Core wants to inherit this lock (you can't call out to it)

Reply:
We could make it not inherit `PermissionsV2`, but that would require virtual function overriding in CoreV2, which I'd rather not do.